x-pack/platform/plugins/shared/fleet/server/tasks/auto_install_content_packages_task.ts

```diff
@@ -295,11 +296,11 @@ export class AutoInstallContentPackagesTask {
             .map((dataset) => `"${dataset}"`)
             .join(',')})`
         : '';
-    const response = await esClient.esql.query({
-      query: `FROM logs-*,metrics-*,traces-* | KEEP @timestamp, data_stream.dataset | WHERE @timestamp > NOW() - ${this.intervalToEsql(
-        this.taskInterval
-      )} | STATS COUNT(*) BY data_stream.dataset | LIMIT 100 ${whereClause}`,
-    });
+    const query = `FROM logs-*,metrics-*,traces-* 
+      | KEEP @timestamp, data_stream.dataset 
+      | WHERE @timestamp > NOW() - ${this.intervalToEsql(this.taskInterval)} 
+      | STATS COUNT(*) BY data_stream.dataset ${whereClause}`;
+    const response = await esClient.esql.query({ query });
```

Comment:
Removed the 100 limit, because there can be much more than 100 datasets when many packages are installed. And even when filtering out datasets from already installed content packages, there might still be many non-content package datasets.